src/libraries/System.Private.CoreLib/src/System/IO/SharedMemoryManager.Unix.cs

```diff
@@ -0,0 +1,835 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO
+{
+    internal readonly unsafe struct SharedMemoryId
+    {
+        private const string UserUnscopedRuntimeTempDirectoryName = ".dotnet";
+
+        private const string UserScopedRuntimeTempDirectoryName = ".dotnet-uid";
+
+        private const string SharedMemoryGlobalDirectoryName = "global";
+
+        private const string SharedMemorySessionDirectoryName = "session";
+        private static int SessionId { get; } = Interop.Sys.GetSid(Environment.ProcessId);
+
+        public SharedMemoryId(string name, bool isUserScope)
+        {
+            if (name.StartsWith("Global\\", StringComparison.Ordinal))
+            {
+                IsSessionScope = false;
+                name = name.Substring("Global\\".Length);
+            }
+            else
+            {
+                IsSessionScope = true;
+                if (name.StartsWith("Local\\", StringComparison.Ordinal))
+                {
+                    name = name.Substring("Local\\".Length);
+                }
+            }
+
+            Name = name;
+
+            if (name.ContainsAny(['\\', '/']))
+            {
+                throw new ArgumentException($"Name '{name}' cannot contain path separators after prefixes.", nameof(name));
```

Comment:
This is IO exception today: IOException: The filename, directory name, or volume label syntax is incorrect. : 'A/X'.

Also, add test?

Reply:
Where is that test? It is not what I have seen in my quick ad-hoc test:
```
jkotas@JKOTAS-STUDIO:~/test$ cat Program.cs
using System.Threading;

new Mutex(false, "A/X");
jkotas@JKOTAS-STUDIO:~/test$ dotnet run
Unhandled exception. System.IO.IOException: The filename, directory name, or volume label syntax is incorrect. : 'A/X'.
   at System.Threading.Mutex.CreateMutexCore(Boolean initiallyOwned, String name, Boolean& createdNew)
   at System.Threading.Mutex..ctor(Boolean initiallyOwned, String name)
   at Program.<Main>$(String[] args) in /home/jkotas/test/Program.cs:line 3
```

Reply:
Looking back at the tests, looks like the length validation is the one that requires `ArgumentException`, not the directory pathing. For whatever reason though, I definitely remember having to add this check to get things to work. I'll go re-validate this.